script/api.py

```diff
@@ -37,36 +37,32 @@ def signup():
     email = request.json.get("email")
     password = request.json.get("password")
 
-    if not (username and email and password):
-        return (
-            jsonify({"message": "either username or email or password is missing"}),
-            400,
-        )
+    if not (username and type(username) == str):
+        return jsonify({"message": "please provide valid username"}), 400
 
     try:
-        validate_email(email)
-    except:
-        return jsonify({"message": "email is not in correct format"}), 400
-
-    if not (type(username) == str):
-        return jsonify({"message": "username is not in correct format"}), 400
+        if not (email and type(email) == str):
+            return jsonify({"message": "please provide valid email"}), 400
+        user_email = validate_email(email).email
+    except EmailNotValidError as error:
+        return jsonify({"message": f"{error}"}), 400
 
-    if not (type(password) == str):
-        return jsonify({"message": "password is not in correct format"}), 400
+    if not (password and type(password) == str):
+        return jsonify({"message": "please provide valid password"}), 400
 
-    cur.execute("SELECT email FROM users WHERE email='{}'".format(email))
+    cur.execute("SELECT email FROM users WHERE email='{}'".format(user_email))
     user_with_email = cur.fetchone()
 
     if user_with_email:
-        return jsonify({"message": "User already exist"}), 400
+        return jsonify({"message": "user already exist"}), 400
 
     cur.execute(
         "INSERT INTO users (name,email,password) VALUES ('{}','{}',crypt('{}',gen_salt('bf')))".format(
-            username, email, password
+            username, user_email, password
         )
     )
```

Comment:
We can combine these two queries into one. If there is a unique index on email field, so whenever we tried to insert same value again, it will give us Duplicate Key error. We can catch that error and report that user already exists.